xarray/test/test_dataarray.py

```diff
@@ -2268,3 +2268,16 @@ def test_dot(self):
             da.dot(dm.values)
         with self.assertRaisesRegexp(ValueError, 'no shared dimensions'):
             da.dot(DataArray(1))
+
+    def test_binary_op_join_setting(self):
+        dim = 'x'
+        align_type = "outer"
+        coords_l, coords_r = [0, 1, 2], [1, 2, 3]
+        missing_0 = xr.DataArray(coords_l, [(dim, coords_l)])
+        missing_3 = xr.DataArray(coords_r, [(dim, coords_r)])
+        with xr.set_options(arithmetic_join=align_type):
+            experimental = missing_0 + missing_3
+            missing_0_aligned, missing_3_aligned =\
+                xr.align(missing_0, missing_3, join=align_type)
+            control = missing_0_aligned + missing_3_aligned
```

Comment:
could also explicitly set `control = xr.DataArray([np.nan, 2, 4, np.nan], [(dim, [0, 1, 2, 3])])`, which is actually even better than using align and arithmetic here